From 546ff775058aa22c1ef20727522dfe1a87b5675d Mon Sep 17 00:00:00 2001
From: G Kranthi <gudishax.kranthikumar@intel.com>
Date: Thu, 30 Jun 2016 14:27:22 +0530
Subject: [PATCH 2075/2508] ASoC: Intel: Skylake: Fix stream tag position in
 purge FW request

commit 9d893e96d1a5cfa46ae29a8199ac7d8be4246b05 from
https://github.com/01org/linux-apollolake-i

Stream tag should be shifted left by 9 bits while inserting
in the purge request.

Change-Id: I62fc56563cff2465710eb4a660b4e891fa2e1663
Signed-off-by: G Kranthi <gudishax.kranthikumar@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/10037
Reviewed-by: Jayanti, Satya Charitardha <satya.charitardha.jayanti@intel.com>
Tested-by: Jayanti, Satya Charitardha <satya.charitardha.jayanti@intel.com>
---
 sound/soc/intel/skylake/bxt-sst.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sound/soc/intel/skylake/bxt-sst.c b/sound/soc/intel/skylake/bxt-sst.c
index fbfdc6b..3de94e4 100644
--- a/sound/soc/intel/skylake/bxt-sst.c
+++ b/sound/soc/intel/skylake/bxt-sst.c
@@ -234,7 +234,7 @@ static int sst_bxt_prepare_fw(struct sst_dsp *ctx, const void *fwdata,
 
 	/* Step 2: Purge FW request */
 	sst_dsp_shim_write(ctx, SKL_ADSP_REG_HIPCI, SKL_ADSP_REG_HIPCI_BUSY |
-					 BXT_IPC_PURGE_FW | (stream_tag - 1));
+					 BXT_IPC_PURGE_FW | (stream_tag - 1) << 9);
 
 	/* Step 3: Unset core0 reset state */
 	ret = skl_dsp_core_unset_reset_state(ctx, SKL_DSP_CORE0_MASK);
-- 
1.7.5.4

